src/Gren/Kernel/Scheduler.js

```diff
@@ -49,6 +49,47 @@ function _Scheduler_receive(callback) {
   };
 }
 
+function _Scheduler_concurrent(tasks) {
+  if (tasks.length === 0) return _Scheduler_succeed([]);
+
+  return _Scheduler_binding(function (callback) {
+    let count = 0;
+    let results = [];
```

Comment:
Nitpick: You can initialize this as `new Array(tasks.length)`. This should pre-initialize the array with the correct length, which I suspect will play a bit better with the JIT. Unlikely to make a huge difference in perf though.

Reply:
I guess the same thing isn't true for `procs` as it's re-assigned in `procs = tasks.map((task, i) => { ...`?

Reply:
We could do:
```
let procs
```
Which initialises to `undefined`